kolibri/plugins/management/assets/src/device_management/views/available-channels-page/channel-token-modal.vue

```diff
@@ -0,0 +1,133 @@
+<template>
+
+  <core-modal
+    :title="$tr('enterChannelToken')"
+    :hideTopButtons="true"
+  >
+    <p>{{ $tr('tokenExplanation') }}</p>
+
+    <form @submit.prevent="submitForm">
+      <ui-alert
+        v-if="tokenNetworkError"
+        :dismissible="false"
+        type="error"
+      >
+        {{ $tr('networkErrorMessage') }}
+      </ui-alert>
+
+      <k-textbox
+        :label="$tr('channelTokenLabel')"
+        v-model.trim="token"
+        :invalid="!tokenIsValid"
+        :invalidText="$tr('invalidTokenMessage')"
+        autofocus
+        @blur="tokenIsBlurred=true"
+        :disabled="formIsDisabled"
+      />
+
+      <div class="buttons">
+        <k-button
+          :text="$tr('cancel')"
+          name="cancel"
+          appearance="flat-button"
+          @click="$emit('closemodal')"
+          :disabled="formIsDisabled"
+        />
+        <k-button
+          class="submit"
+          type="submit"
+          :text="$tr('confirm')"
+          :primary="true"
+          :disabled="formIsDisabled"
+        />
+      </div>
+    </form>
+  </core-modal>
+
+</template>
+
+
+<script>
+
+  import UiAlert from 'keen-ui/src/UiAlert';
+  import coreModal from 'kolibri.coreVue.components.coreModal';
+  import kButton from 'kolibri.coreVue.components.kButton';
+  import kTextbox from 'kolibri.coreVue.components.kTextbox';
+  import { getRemoteChannelByToken } from '../../state/actions/availableChannelsActions';
+
+  export default {
+    name: 'channelTokenModal',
+    components: {
+      UiAlert,
+      coreModal,
+      kButton,
+      kTextbox,
+    },
+    data() {
+      return {
+        formIsDisabled: false,
+        formIsSubmitted: false,
+        token: '',
+        tokenIsBlurred: false,
+        tokenLookupFailed: false,
+        tokenNetworkError: false,
```

Comment:
Interesting bringing a lot of this into local component state.

I think I like it. Curious to hear about whether you also considered putting it in Vuex. In vuex 2 with modules and namespacing, would we be more likely to put it there?